tweak s3db timings

dannyroberts opened the pull request.

dannyroberts committed:
Tweak S3BlobDB timing buckets

dannyroberts committed:
Send long S3BlobDB requests to sentry

millerdev commented on corehq/blobs/s3db.py:
```diff
@@ -41,10 +42,16 @@ def __init__(self, config):
+        def record_long_request(duration):
+            if duration > 30:
+                notify_exception(None, "S3BlobDB request took a long time.", details={
+                    'duration': duration,
```
What do you think of adding more details here like the blob id, blob bucket, action, and possibly blob size (if known)? I'm curious how this alert will be useful or actionable if it doesn't include identifying information about the blob.

dannyroberts replied:
Yeah that's definitely a good call. I think I'll also bump the threshold up to something even more absurd like 100s so we get fewer notifications. We have already seen a handful of requests come in at over 100s so those would be a good place to start getting more granular info on.

millerdev replied:
The specific blob that took a long time cannot be identified yet, and I think that info would be very helpful in investigating why it took a long time. The blob identifier and bucket (blobdb bucket, not s3 bucket) are available everywhere `report_timing` is called. What do you think of changing the signature to `def report_timing(action, identifier, bucket):`? That would allow you to add those things to the details here.

millerdev reviewed (commented):
I'd like to be convinced that we actually want the ~alert~ `notify_exception` and that it will not just add more non-actionable noise to sentry.

dannyroberts committed:
Make S3BlobDB long request notification less frequent and more useful

dannyroberts commented:
I also bumped boto log levels up to INFO so that S3 request info appears in breadcrumbs. I'll check out what it looks like on staging for a bit before merging, to make sure it's not chattier than we need.

dannyroberts committed:
Change boto log level back to WARNING

INFO is way too chatty, and we already get logging from requests
which covers everything we need

dannyroberts commented:
Okay, yeah INFO is too chatty, and I found that `requests` logging actually already covers the basic info we need like what the blob path is.

dannyroberts commented:
@millerdev I was saying that sentry already collects this info in breadcrumbs because we capture `requests` logging and boto uses `requests`, so we can see what URL the S3 request is made to. I can still add that info though if you think it'd be helpful to make more prominent.

dannyroberts committed:
Add blobdb_bucket and identifier info to long s3db request logging

millerdev reviewed (approved):
> sentry already collects this info in breadcrumbs

Ah, I did not catch that in your earlier comments. Thanks for the clarification.
